### PR TITLE
ARROW-14049: [C++][Java] Upgrade ORC to 1.7.0

### PR DESCRIPTION
**Pull request by williamhyun**
This PR aims to bump ORC to version 1.7.0.

[Apache ORC 1.7.0](https://orc.apache.org/news/2021/09/15/ORC-1.7.0/) is a new release with the following new features and improvements.

ORC-377 Support Snappy compression in C++ Writer
ORC-577 Support row-level filtering
ORC-716 Build and test on Java 17-EA
ORC-731 Improve Java Tools
ORC-742 LazyIO of non-filter columns
ORC-751 Implement Predicate Pushdown in C++ Reader
ORC-755 Introduce OrcFilterContext
ORC-757 Add Hashtable implementation for dictionary
ORC-780 Support LZ4 Compression in C++ Writer
ORC-797 Allow writers to get the stripe information
ORC-818 Build and test in Apple Silicon
ORC-861 Bump CMake minimum requirement to 2.8.12
ORC-867 Upgrade hive-storage-api to 2.8.1
ORC-984 Save the software version that wrote each ORC file

### CODE REVIEW

File: cpp/thirdparty/versions.txt [hunk 1]
```diff
@@ -60,7 +60,7 @@ ARROW_LZ4_BUILD_SHA256_CHECKSUM=030644df4611007ff7dc962d981f390361e6c97a34e5cbc3
 # https://github.com/microsoft/mimalloc/issues/353
 ARROW_MIMALLOC_BUILD_VERSION=v1.7.2
 ARROW_MIMALLOC_BUILD_SHA256_CHECKSUM=b1912e354565a4b698410f7583c0f83934a6dbb3ade54ab7ddcb1569320936bd
-ARROW_ORC_BUILD_VERSION=1.6.9
+ARROW_ORC_BUILD_VERSION=1.7.0
 ARROW_ORC_BUILD_SHA256_CHECKSUM=2b79f1152454a38abd7b361b407f65689e275f1fb28d4293892e42fbebcc0a9d
```
Comment (dongjoon-hyun) [inline cpp/thirdparty/versions.txt:64]: Hi, @williamhyun .
Could you update this line together?
```
$ shasum -a256 < /Users/dongjoon/APACHE/arrow/cpp/thirdparty/orc-1.7.0.tar.gz
45d6ba9149ffa2aaa168d61ab326f61181861c94529f26da3918a9aa2f801e39
```

FYI, the previous value came from here.
```
$ shasum -a256 < /Users/dongjoon/APACHE/arrow/cpp/thirdparty/orc-1.6.9.tar.gz
2b79f1152454a38abd7b361b407f65689e275f1fb28d4293892e42fbebcc0a9d  -
```

**Comment by williamhyun** [comment]
Thank you @dongjoon-hyun !

I updated my PR.

**Comment by pitrou** [comment]
Sidenote: the [vcpkg port](https://vcpkg.info/port/orc) would deserve updating as well (it's used for building our Python binary wheels, for example).

**Comment by pitrou** [comment]
The two test failures above are unrelated, I'll merge this.

**Comment by dongjoon-hyun** [comment]
Thank you so much, @williamhyun and @pitrou !

**Comment by dongjoon-hyun** [comment]
> Sidenote: the [vcpkg port](https://vcpkg.info/port/orc) would deserve updating as well (it's used for building our Python binary wheels, for example).

BTW, thank you for the tip. Apache ORC community was aware of the existence of `vcpkg port` until Today. We will take a look.

**Comment by williamhyun** [comment]
Thank you @pitrou and @dongjoon-hyun !

**Pull request closed without merging**